packages/react/time-series/src/utils.ts

```diff
@@ -217,3 +218,74 @@ export function updateChartConfig(options: UpdateChartConfig) {
 
   chart.options.plugins = customPlugins
 }
+
+interface GetScalesOptions {
+  styles?: Styles
+  granularity: TimeSeriesGranularity | null
+  isFormatted: boolean
+  isStatic: boolean
+}
+
+export function getScales(options: GetScalesOptions) {
+  const { styles, granularity, isFormatted, isStatic } = options
+  const scale = styles?.yAxis?.scale || defaultStyles.yAxis.scale
+
+  const hideGridLines = styles?.canvas?.hideGridLines || defaultStyles.canvas.hideGridLines
+
+  const scalesBase = {
+    x: {
+      display: !hideGridLines,
+      grid: {
+        drawOnChartArea: false
+      },
+      beginAtZero: true
+    },
+    y: {
+      display: !hideGridLines,
+      beginAtZero: styles?.yAxis?.beginAtZero || defaultStyles.yAxis.beginAtZero,
+      grid: { drawOnChartArea: true }
+    }
+  }
+
+  const customFormatScales = {
+    ...scalesBase
+  }
+
+  const autoFormatScales = {
+    ...scalesBase,
+    x: {
+      ...scalesBase.x,
+      type: 'timeseries',
+      time: {
+        unit: getGranularityBasedUnit(granularity)
+      }
+    }
+  }
+
+  const currentFormatScales = isFormatted || isStatic ? customFormatScales : autoFormatScales
+
+  const linearScales: DeepPartial<{ [key: string]: ScaleOptionsByType<'linear'> }> = {
+    ...currentFormatScales,
+    y: {
+      ...currentFormatScales.y,
+      type: 'linear'
+    }
+  }
+
+  const logarithmicScales: DeepPartial<{ [key: string]: ScaleOptionsByType<'logarithmic'> }> = {
+    ...(isFormatted || isStatic ? customFormatScales : autoFormatScales),
+    y: {
+      ...currentFormatScales.y,
+      type: 'logarithmic'
+    }
+  }
+
+  return {
+    linear: {
+      ...linearScales
+    },
+    logarithmic: {
+      ...logarithmicScales
+    }
+  }[scale] as ChartScales
```

Comment:
These are new. BTW, is it simpler to do

```ts
  return scale === 'linear'
    ? linearScales
    : logarithmicScales
```

Also, you could even move the computation of `linearScales` and `logarithmicScales` into an `if`-statement. For example,

```ts
if (scale === 'linear') {
  return {
    /* … */
  }
} else {
  return {
    /* … */
  }
}
```

Reply:
Yes, I think an if statement will work better here, refactoring